src/server/storage/storageAPI.js

```diff
@@ -28,29 +27,37 @@ const setup = (app: Router, gunPublic: StorageAPI, storage: StorageAPI) => {
     passport.authenticate('jwt', { session: false }),
     wrapAsync(async (req, res) => {
       const { body, user: userRecord } = req
+      const { user: userPayload } = body
       const logger = req.log
-      logger.debug('new user request:', { data: body.user, userRecord })
 
-      //if torus, then we first verify the user mobile/email by verifying it matches the torus public key
-      //(torus maps identifier such as email and mobile to private/public key pairs)
-      const { torusProof, torusProvider, torusProofNonce, email, mobile, ...bodyUser } = body.user || {}
-      if (torusProof) {
-        const { emailVerified, mobileVerified } = await TorusVerifier.verifyProof(
-          torusProof,
-          torusProvider,
-          body.user,
-          torusProofNonce
-        ).catch(e => {
-          logger.warn('TorusVerifier failed:', { e, msg: e.message })
-          return { emailVerified: false, mobileVerified: false }
-        })
+      logger.debug('new user request:', { data: userPayload, userRecord })
+
+      const {
+        torusProof,
+        torusProvider,
+        torusProofNonce,
+        torusAccessToken,
+        torusIdToken,
+        email,
+        mobile,
+        ...payloadWithoutCreds
+      } = userPayload || {}
+
+      // if torus, then we first verify the user mobile/email by verifying it matches the torus public key
+      // (torus maps identifier such as email and mobile to private/public key pairs)
+      if (torusProof || torusAccessToken) {
+        const verifier = createUserVerifier(userRecord, userPayload, logger)
 
-        logger.info('TorusVerifier result:', { emailVerified, mobileVerified })
-        userRecord.smsValidated |= mobileVerified
-        userRecord.isEmailConfirmed |= emailVerified
+        if (torusProvider == 'facebook') {
```

Comment:
it was better when it was encapsulated inside the verifier

Reply:
generally we have now 2 totally different ways:
- via proof
- via access token

they have different input params and different result
so they couldn't be processed using the single service class (e.g. TorusVerifier)

I've added ./verifier that incapsulates both flows.
We could move torusProvider == 'facebook' check inside